docs/decisions/ADR-0001/DECISION.md

```
@@ -0,0 +1,95 @@
+
+    App->>Core: create_commit(parent, tree, signature)
+    Core->>Core: commit = Commit { ... }
+    Core->>Core: hash = compute_commit_id(&commit)
+    Core->>Store: put_object(hash, serialize(commit))
+    Store-->>Core: (persists object)
+    Core-->>App: return hash
+```
+
+This approach provides maximum portability and a lower attack surface for the core logic, without sacrificing the ability to use standard Git tooling on host systems.
+
+## Consequences
```

markdownlint-cli2 (0.18.1)

Trailing spaces
Expected: 0 or 2; Actual: 1

(MD009, no-trailing-spaces)